src/mixtures/mixturemodel.jl

```diff
@@ -334,7 +335,7 @@ function _mixlogpdf!(r::AbstractArray, d::AbstractMixtureModel, x)
 
             # in the mean time, add log(prior) to lp and
             # update the maximum for each sample
-            for j = 1:n
+            for j in eachindex(r)
```

Comment:
This seems problematic as `j` is used to index into `lp_i` which is a view of a `Matrix` column. We should use two different indices or `for (j, lp_ij) in zip(eachindex(m), lp_i)` as below. `r` isn't relevant here anyway.